pkg/cmd/roachtest/roachtestutil/mixedversion/runner.go

```diff
@@ -613,7 +632,10 @@ func (tr *testRunner) refreshServiceData(ctx context.Context, service *serviceRu
 		return err
 	}
 
-	tr.monitor.Init()
+	if tr.plan.deploymentMode == SeparateProcessDeployment && service.descriptor.Name != install.SystemInterfaceName {
+		tr.monitor.Init()
```

Comment:
Who monitors the system tenant now in case of shared-process and single-tenant?

Reply:
Yikes, good catch! The intended logic is to only start the monitor once every cockroach process is running; in the case of separate-process, this is equivalent to checking if the _tenant_ is running (since system is always started first).

Fixed.